application/src/main/java/com/github/pieter_groenendijk/model/DigitalProduct.java

```diff
@@ -0,0 +1,66 @@
+package com.github.pieter_groenendijk.model;
+
+import jakarta.persistence.*;
+
+import java.util.Date;
+
+@Entity
+@Table(name = "DigitalProduct")
+public class DigitalProduct extends Product {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long digitalProductId;
+
+    @Column(name = "title", nullable = false, length = 100)
+    private String accessStatus;
+
+    @Column(name = "releaseDate", nullable = true)
+    private Date releaseDate;
+
+    @Column(name = "classification", nullable = true)
+    private int classification;
+
+    @Column(name = "mediaType", nullable = false)
+    private String mediaType;
```

Comment:
Het idee was dat we dus ook verschillende media types gingen aanbieden: ebooks, films, series? Bevat elk van zo'n digitaal product dezelfde informatie?

Reply:
Eens, dit geld ook voor fysieke producten.
Een DVD heeft niet dezelfde eigenschappen als een fysiek boek. Het zat idd in mn achterhoofd te dwalen hoe dit het beste te abstraheren naar 1 template soort.  Maar ik denk eigenlijk:Template Method volgen voor Product, een template digital en template physical en dan kunnen de individuele soorten ook eigen eigenschappen hebben.

En als ik er te lang over nadenk kan het ook een Strategy worden of een Factory, maar for now wat Template mn eerst ingeving :)

Reply:
Klinkt goed.

Reply:
Ik brainstorm even hardop
AudioProduct: audiobooks/muziek
VideoProduct: series, films, ect
ReadProduct: ebooks, magazines
and they then extend DigitalProduct

Reply:
Je hebt ook nog "Table Per Class" Inheritance. Ik denk dat de twijfel tussen "Single Table" of "Table Per Class" zou moeten zitten.

Reply:
Opzich klinkt de inheritance structuur die je beschrijft theoretisch wel goed. In de praktijk zullen we het nog uit moeten vinden. Ook omdat er voor mij nog best veel vraagtekens zijn, bijv. streaming zooi. Niet elk digitaal product zal gestreamt worden (ebooks bijv.).

Echter denk ik dat we juist daarom nu naar een prototype hiervan moeten gaan. Dan komen we ervan zelf achter hoe dat gaat werken. Dus binnenkort een keer ff samen over streaming na gaan denken?

Reply:
En nu je het inderdaad zegt. Lijkt mij inderdaad ook niet dat abstracties hun eigen id (bijv. digitalProductId) nodig hebben.